ext/include/opentelemetry/ext/http/client/http_client_factory.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+#include "opentelemetry/ext/http/client/http_client.h"
+#ifdef WITH_CURL
+#  include "opentelemetry/ext/http/client/curl/http_client_curl.h"
+#endif
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace ext
+{
+namespace http
+{
+namespace client
+{
+
+class HttpClientFactory
+{
+public:
+#ifdef WITH_CURL
+  static std::shared_ptr<SessionManager> Create()
+  {
+    return std::make_shared<ext::http::client::curl::SessionManager>();
+  }
+#else
+  static std::shared_ptr<SessionManager> Create()
+  {
+    return std::shared_ptr<SessionManager>(nullptr);
```

Comment:
is it better to fail compilation if no HTTP library is wired in?

Reply:
Correct. This should be the right behavior. Will change.

Reply:
This is done now. `HttpClientFactory::Create` method will now be missing if no HTTP library is plugged in.